sos/policies/__init__.py

```diff
@@ -49,6 +51,107 @@ def load(cache={}, sysroot=None):
     return cache['policy']
 
 
+class ContainerRuntime(object):
+    """Encapsulates a container runtime that provides the ability to plugins to
+    check runtime status, check for the presence of specific containers, and
+    to format commands to run in those containers
+    """
+
+    runtime_name = 'Undefined'
+    containers = []
+    images = []
+    volumes = []
+    binary = ''
+
+    def __init__(self, policy=None):
+        self.policy = policy
+        self.run_cmd = "%s exec " % self.binary
+
+        self.active = self._check_is_active()
+        self.containers = self.get_containers()
+        self.images = self.get_images()
+        self.volumes = self.get_volumes()
+
+    def _check_is_active(self):
+        """Check to see if the container runtime is both present AND active.
+
+        Active in this sense means that the runtime can be used to glean
+        information about the runtime itself and containers that are running.
+        """
+        return is_executable(self.binary)
+
+    def get_containers(self, get_all=False):
+        """Get a list of containers present on the system.
```

Comment:
Maybe worth noticing here that the list contains tuples `(conatiner_id, container_name)` - it took me a while (and a "podman ps" output) to understand it from the `containers.append((ent[0], ent[-1]))` line.

Reply:
Done.